verification/truststore/truststore.go

```diff
@@ -0,0 +1,125 @@
+// Package truststore reads certificates in a trust store
+package truststore
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	corex509 "github.com/notaryproject/notation-core-go/x509"
+	"github.com/notaryproject/notation-go/dir"
+)
+
+// Type is an enum for trust store types supported such as
+// "ca" and "signingAuthority"
+type Type string
+
+const (
+	TypeCA               Type = "ca"
+	TypeSigningAuthority Type = "signingAuthority"
+)
+
+var (
+	Types = []Type{
+		TypeCA,
+		TypeSigningAuthority,
+	}
+)
+
+// X509TrustStore provide list and get behaviors for the trust store
+type X509TrustStore interface {
+	// GetCertificates returns certificates under storeType/namedStore
+	GetCertificates(ctx context.Context, storeType Type, namedStore string) ([]*x509.Certificate, error)
+}
+
+// NewX509TrustStore generates a new X509TrustStore
+func NewX509TrustStore(trustStorefs dir.SysFS) X509TrustStore {
+	return x509TrustStore{trustStorefs}
+}
+
+// x509TrustStore implements X509TrustStore
+type x509TrustStore struct {
+	trustStorefs dir.SysFS
+}
+
+// GetCertificates returns certificates under storeType/namedStore
+func (trustStore x509TrustStore) GetCertificates(ctx context.Context, storeType Type, namedStore string) ([]*x509.Certificate, error) {
+	if storeType == "" || namedStore == "" {
```

Comment:
IMO we should restrict most, if not all, the user-inputs to the following regex [a-zA-Z0-9_.-]+ i.e. alphanumeric and _, -.
This will avoid most of the [cross-platform issues](https://superuser.com/questions/358855/what-characters-are-safe-in-cross-platform-file-names-for-linux-windows-and-os). Also, we can either update spec with this PR or later(please open a issue, if we choose to do it later).


https://github.com/notaryproject/notation/pull/405#discussion_r1009061509